applications/tari_validator_node/tests/cucumber.rs

```diff
@@ -885,24 +885,73 @@ async fn works_indexer_graphql(world: &mut TariWorld, indexer_name: String) {
     // insert event mock data in the substate manager database
     indexer.insert_event_mock_data().await;
     let mut graphql_client = indexer.get_graphql_indexer_client().await;
+    let component_address = [0u8; 32];
     let template_address = [0u8; 32];
     let tx_hash = [0u8; 32];
     let query = format!(
-        "{{ getEvent(templateAddress: {:?}, txHash: {:?}) {{ templateAddress, txHash, topic, payload }} }}",
-        template_address, tx_hash
+        "{{ getEventsForTransaction(txHash: {:?}) {{ componentAddress, templateAddress, txHash, topic, payload }}
+    }}",
+        tx_hash.to_hex()
     );
     let res = graphql_client
         .send_request::<HashMap<String, Vec<tari_indexer::graphql::model::events::Event>>>(&query, None, None)
         .await
-        .unwrap();
-    let res = res.get("getEvent").unwrap();
+        .expect("Failed to obtain getEventsForTransaction query result");
+    let res = res.get("getEventsForTransaction").unwrap();
     assert_eq!(res.len(), 1);
-    assert_eq!(res[0].template_address, [0u8; 32]);
-    assert_eq!(res[0].tx_hash, [0u8; 32]);
+    assert_eq!(res[0].component_address, Some(component_address));
+    assert_eq!(res[0].template_address, template_address);
+    assert_eq!(res[0].tx_hash, tx_hash);
     assert_eq!(res[0].topic, "my_event");
     assert_eq!(res[0].payload, HashMap::from([("my".to_string(), "event".to_string())]));
 }
 
+#[when(expr = "indexer {word} scans the network {int} events for account {word} with topics {word}")]
+async fn indexer_scans_network_events(
+    world: &mut TariWorld,
+    indexer_name: String,
+    num_events: u32,
+    account_name: String,
+    topics: String,
+) {
+    let indexer: &mut IndexerProcess = world.indexers.get_mut(&indexer_name).unwrap();
+    let accounts_component_addresses = world.outputs.get(&account_name).expect("Account name not found");
+    let component_addresses = accounts_component_addresses
+        .into_iter()
+        .filter(|(k, _)| k.contains("components/Account"))
+        .map(|(_, v)| {
+            v.address
+                .as_component_address()
+                .expect("Failed to parse `ComponentAddress`")
+        })
+        .collect::<Vec<_>>();
+
+    let component_address = *component_addresses
+        .first()
+        .unwrap_or_else(|| panic!("Did not find component addresses"));
```

Comment:
nit: If you are only interested in finding a single item, you can make use of `Iterator::find` without having to collect all matches first

```suggestion
    let component_address = accounts_component_addresses
        .into_iter()
        .find(|(k, _)| k.contains("components/Account"))
        .map(|(_, v)| {
            v.address
                .as_component_address()
                .expect("Failed to parse `ComponentAddress`")
        })
       .expect("Did not find component addresses");

```